dot/core/service.go

```diff
@@ -317,6 +323,7 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 	}
 
 	// Check transaction validation on the best block.
+	// TODO does this need to be updated like txnPool did
 	rt, err := s.blockState.GetRuntime(nil)
```

Comment:
Should this be updated to take in a blockHash similar to the change to maintain transaction pool?

Reply:
yes i do! sweet sweet